api/discovery/discovery.go

```diff
@@ -73,6 +101,36 @@
 		GeoLocation:  location,
 		Type:         typ,
 		Labels:       labels,
+		Parent:       parent,
 		Raw:          rawString,
 	}
 }
+
+func (r *Resource) ToVocResource() (voc.IsCloudResource, error) {
+	var (
+		b   []byte
+		err error
+	)
+
+	typ := strings.Split(r.ResourceType, ",")[0]
```

Comment:
Thus, we have to keep in mind that 1st) we have to use the right order of types in r.ResourceType (e.g., VirtualMachine before Compute) and 2nd) they have to accurately match the name of the structs in voc

Reply:
That should be guaranteed by owl2java. My idea would also to auto-generate this function at some point.